test/distributed/org/apache/cassandra/distributed/impl/Instance.java

```diff
@@ -599,20 +602,61 @@ public int liveMemberCount()
     public NodeToolResult nodetoolResult(boolean withNotifications, String... commandAndArgs)
     {
         return sync(() -> {
-            DTestNodeTool nodetool = new DTestNodeTool(withNotifications);
-            int rc =  nodetool.execute(commandAndArgs);
-            return new NodeToolResult(commandAndArgs, rc, new ArrayList<>(nodetool.notifications.notifications), nodetool.latestError);
+            try (CapturingConsoleOutputProvider provider = new CapturingConsoleOutputProvider())
+            {
+                DTestNodeTool nodetool = new DTestNodeTool(withNotifications, provider);
+                int rc = nodetool.execute(commandAndArgs);
+                return new NodeToolResult(commandAndArgs, rc,
+                                          new ArrayList<>(nodetool.notifications.notifications),
+                                          nodetool.latestError,
+                                          provider.getOutString(),
+                                          provider.getErrString());
+            }
         }).call();
     }
 
+    private static class CapturingConsoleOutputProvider implements ConsoleOutputProvider
+    {
+        private final ByteArrayOutputStream outBase = new ByteArrayOutputStream();
```

Comment:
this will cause a compiler warning, will need something like `@SuppressWarnings("resources)` to avoid